app/scripts/services/dimStoreService.factory.js

```diff
@@ -1593,5 +1602,19 @@
       return ret;
     }
     // code above is from https://github.com/DestinyTrialsReport
+
+    function loadJSON(file) {
+      return $http.get(file)
+               .then(function(json) {
+                 return json.data;
+               });
+    }
+
+    function getMissingSourceHashes(itemHash, missingSources) {
```

Comment:
I think `missingSources` will be a promise here, so you need to use `then` to asynchronously access its data. Best to pass `missingSources` into one of the `$q.all` resolvers like the one that takes `dimBucketService` and `dimDefinitions` up top.